packages/dd-trace/src/instrumenter.js

```diff
@@ -81,7 +81,8 @@ class Instrumenter {
       Object.keys(plugins)
         .filter(name => !this._plugins.has(plugins[name]))
         .forEach(name => {
-          this._set(plugins[name], { name, config: getConfig(name) })
+          const options = config.serviceMapping[name] ? { service: config.serviceMapping[name] } : {}
```

Comment:
The mapping should work on the service name and not on the plugin name.

Reply:
The mapping should also be tested.

Reply:
I am currently setting `export DD_SERVICE_MAPPING=fs:service-name,redis:service-name` where key is the plugin name. How should I change it to?

Reply:
It should be using the service name and not the plugin name. So for example, for a service named `orders` that uses MySQL, you can look at the `mysql2` plugin and see that the end result is `orders-mysql`. So to rename that service, you would do `orders-mysql:somethine-else`. This is a generic logic that applies to any span and not just those generated by auto-instrumentation, so it should also not go in the instrumenter since that's only used by plugins.

Reply:
Understood. Can you point me to appropriate files where I have to make the change?

I was using `export DD_SERVICE_MAPPING=fs:service-name,redis:service-name` style of logic for java services, but what you said makes sense for nodejs apps.

Reply:
Actually Java does the same thing, but the main difference is the default service for MySQL is `mysql` as opposed to Node that uses the service name of the actual service as a prefix.

As for which file, it could be in many places, I didn't really take the time to look into it. The formatter could be one, but I don't think it has access to the config. It has to not be too early since the service name could be changed programmatically, and I'd expect that the final name is the one that would be mapped (correct me if you think that behaviour is wrong).

Reply:
Will try this in a few days and let you know. Currently occupied with datadog integration internally.